src/ast.fs

```diff
@@ -193,7 +193,7 @@ type Shader = {
     code: TopLevel list
     forbiddenNames: string list
     reorderFunctions: bool  // set to true if we saw a forward declaration
-} with member this.mangledFilename = (System.IO.Path.GetFileName this.filename).Replace(".", "_")
+} with member this.mangledFilename = (System.IO.Path.GetFileName this.filename).Replace(".", "_").Replace(" ", "_")
```

Comment:
We could replace any non alphanumeric character with `_`. I suppose`-` and others can cause problems.